Sources/Views/InputTextView.swift

```diff
@@ -303,26 +303,50 @@ open class InputTextView: UITextView {
     private func parseForComponents() -> [Any] {
         
         var components = [Any]()
-        let range = NSRange(location: 0, length: attributedText.length)
-        attributedText.enumerateAttributes(in: range, options: []) { (object, range, _) in
-            
-            if object.keys.contains(.attachment) {
-                if let attachment = object[.attachment] as? NSTextAttachment {
-                    if let image = attachment.image {
-                        components.append(image)
-                    } else if let image = attachment.image(forBounds: attachment.bounds,
-                                                           textContainer: nil,
-                                                           characterIndex: range.location) {
-                        components.append(image)
+        var attachments = [(NSRange, UIImage)]()
+        let length = attributedText.length
+        let range = NSRange(location: 0, length: length)
+        attributedText.enumerateAttribute(.attachment, in: range) { (object, range, _) in
+            if let attachment = object as? NSTextAttachment {
+                if let image = attachment.image {
+                    attachments.append((range, image))
+                } else if let image = attachment.image(forBounds: attachment.bounds,
+                                                       textContainer: nil,
+                                                       characterIndex: range.location) {
+                    attachments.append((range,image))
+                }
+            }
+        }
+        
+        var curLocation = 0
```

Comment:
@zhongwuzw Can you help me understand the code from here? I just want to understand before we merge if Nathan doesn't have time to look at this one

Reply:
@SD10 , First we should enumerate all `.attachment`, get its image and range, then we can split text and image based on the range.